core/chains/evm/logpoller/log_poller.go

```diff
@@ -612,12 +617,79 @@ func (lp *logPoller) LatestLogEventSigsAddrsWithConfs(fromBlock int64, eventSigs
 
 // GetBlocks tries to get the specified block numbers from the log pollers
 // blocks table. Returns the blocks it was able to find, empty slice if none.
-// It is not guaranteed to find the blocks specified, since
-// the log poller does not save blocks when backfilling.
-// It can be used as a cache of recent blocks (falling back to an RPC on cache miss),
-// to speed up querying and avoid loading an EVM node.
-func (lp *logPoller) GetBlocks(numbers []uint64, qopts ...pg.QOpt) ([]LogPollerBlock, error) {
-	return lp.orm.GetBlocks(numbers, qopts...)
+// When the log poller does not have requested blocks, it falls back
+// to RPC to fetch the missing blocks.
+// response contains blocks in the same order as "numbers" in request parameters
+func (lp *logPoller) GetBlocks(ctx context.Context, numbers []uint64, qopts ...pg.QOpt) ([]LogPollerBlock, error) {
```

Comment:
shouldn't need the extra context, can be passed in via qopts

Reply:
hmm the context object is for RPC calls. Aren't `qopts` only for interacting with db?

Reply:
that being said, I think if client passes in both context and qopts, I believe we should honor both for the db call. This change is currently missing

Reply:
yeah I was hoping to have a single ctx just so you could cancel it and have it return whether it was doing the DB call or the RPC call. Looking closer I don't see a simple way to just extract a ctx from qopts though (may not be supplied for instance). I think the best you can do is keep the signature as you have it and include a pg.WithContext(ctx) as a qopt in the db call

Reply:
Yep that makes sense. updated